cli/mender-artifact/modify.go

```diff
@@ -89,6 +94,14 @@ func modifyArtifact(c *cli.Context) error {
 	return nil
 }
 
+func cleanSDImg(partitions []partition, candidateType CandidateType) {
+	if candidateType == RawSDImage && len(partitions) == 4 {
```

Comment:
Instead of splitting in two places the knowledge of "len 4 in partitions means this or that", I would put the cleanup part at line 63, like:
```
} else if candidateType == RawSDImage {
    if len(modifyCandidates) == 4 { // sdimg
        modifyCandidates = modifyCandidates[1:3]
        // HERE: defer cleanup
    }
}
```

It would then be more consistent to do an specific cleanup according to which candidateType you got from getCandidatesForModify